trustfall_core/src/interpreter/execution.rs

```diff
@@ -241,6 +241,24 @@ fn construct_outputs<'query, AdapterT: Adapter<'query>>(
     }))
 }
 
+/// takes a FieldValue and produces a usize clamped to 0 and usize::MAX
+fn usize_from_field_value(field_value: &FieldValue) -> Option<usize> {
+    match field_value {
+        FieldValue::Int64(num) => {
+            Some(usize::try_from(*num.max(&0)).expect("i64 can be converted to usize"))
+        }
+        FieldValue::Uint64(num) => {
+            Some(usize::try_from(*num).expect("i64 can be converted to usize"))
```

Comment:
usize should be bigger than both of these right? there's no need to also .min this with `usize::MAX` I think

Reply:
It's not actually guaranteed to, it's possible that `usize` is 32-bit e.g. on `wasm32` platforms whereas `u64` is always 64-bit. Your call on whether we sort this out now or not.

Reply:
But either update the docstring or the code, because right now the docstring says this is clamped to `usize::MAX` when the code doesn't actually clamp it.

Reply:
I update3d the docstring for now to just not say it's clamped to usize::MAX, instead it just says that we return a positive usize.